pkg/scheduling/pd/scheduler.go

```diff
@@ -240,12 +240,11 @@ func (s *Scheduler) generateSchedulerConfig(ctx context.Context, pluginsConfig m
 		}
 	}
 
-	return scheduling.NewSchedulerConfig(
-		preSchedulePlugins,
-		filters,
-		scorers,
-		picker.NewMaxScorePicker(),
-		postSchedulePlugins,
-		postResponsePlugins,
-	)
+	return scheduling.NewSchedulerConfig().
+		WithPreSchedulePlugins(preSchedulePlugins...).
+		WithFilters(filters...).
```

Comment:
q: it is ok to pass an empty array to `With...()`?

Reply:
theoretically, yes. but this would be redundant. 
the `NewSchedulerConfig` default behavior is to put empty arrays.
WithFilters overrides the existing config filters and replaces the existing ones with the gives ones.
<img width="881" alt="image" src="https://github.com/user-attachments/assets/65b4d434-366b-4672-ba2f-c98b4678461e" />